test/server.test.js

```diff
@@ -0,0 +1,80 @@
+const insertImages = require('../database-mongodb/insert_images.js');
+const Images = require('../database-mongodb/Images.js');
+const mongoose = require('mongoose');
+const request = require('supertest');
+const app = require('../server/index.js');
+
+beforeAll(() => {
+  return mongoose.connect(process.env.MONGO_URL, { useNewUrlParser: true })
+    .then(() => {
+      return Images.deleteAll()
+    })
+    .then(() => {
+      let imageDataForSingleItem =
+      {
+        itemId: '1000',
+        itemImages: [{
+          small: 'https://images.unsplash.com/photo-1?w=54',
+          medium: 'https://images.unsplash.com/photo-1?w=400',
+          large: 'https://images.unsplash.com/photo-1?w=1000'
+        },
+        {
+          small: 'https://images.unsplash.com/photo-2?w=54',
+          medium: 'https://images.unsplash.com/photo-2?w=400',
+          large: 'https://images.unsplash.com/photo-2?w=1000'
+        }]
+      }
+      return Images.insertRecords([imageDataForSingleItem])
+    })
+})
+
+afterAll(() => {
+  return mongoose.connection.close();
+})
+
+test('successfully retrieves image data for item 1000', () => {
+  return request(app).get('/itemImages/1000')
+    .then((response) => {
+      expect(response.status).toBe(200);
+      expect(response.body.itemId).toBe('1000');
+      expect(response.body.itemImages.length).toBe(2);
+      expect(response.body.itemImages[0].small).toBe('https://images.unsplash.com/photo-1?w=54');
+      expect(response.body.itemImages[0].medium).toBe('https://images.unsplash.com/photo-1?w=400');
+      expect(response.body.itemImages[0].large).toBe('https://images.unsplash.com/photo-1?w=1000');
+      expect(response.body.itemImages[1].small).toBe('https://images.unsplash.com/photo-2?w=54');
+      expect(response.body.itemImages[1].medium).toBe('https://images.unsplash.com/photo-2?w=400');
+      expect(response.body.itemImages[1].large).toBe('https://images.unsplash.com/photo-2?w=1000');
+    })
+});
+
+test('successfully retrieves main image url for item 1000', () => {
+  return request(app).get('/itemImages/1000/mainImage')
+    .then((response) => {
+      expect(response.status).toBe(200);
+      expect(response.body.image).toBe('https://images.unsplash.com/photo-1?w=54')
+    })
+});
+
+test('returns 404 status code if item not found', () => {
+  return request(app).get('/itemImages/200')
+    .then((response) => {
+      expect(response.status).toBe(404);
+    })
+});
+
+test('returns 404 status code if main image not found', () => {
+  return request(app).get('/itemImages/200/mainImage')
+    .then((response) => {
+      expect(response.status).toBe(404);
+    })
+});
+
+test('successfully removes image data for item 1000', () => {
+  return request(app).delete('/itemImages/1000')
+    .then(() => {
+      request(app).get('/itemImages/1000')
+        .then((response) => {
+          expect(response.status).toBe(404);
+        })
+    })
```

Comment:
hey just as a suggestion, you may want to use some of the lower 0-99 values for testing your 404's.  Because we'll end up scaling up our item amounts, these values above 200 will become available items.  I'm going to have to change some of my tests as well.

Reply:
Good call! I'll do that.